shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/PlantsRepository.kt

```diff
@@ -1,16 +1,55 @@
 package com.vml.tutorial.plantshop.plants.data
 
+import com.vml.tutorial.plantshop.core.utils.Logger
+import com.vml.tutorial.plantshop.plants.domain.DbDataSource
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
 
 interface PlantsRepository {
     fun getPlants(): List<Plant>
+    fun getFavorites(): Flow<List<Plant>>
+    suspend fun toggleFavoriteStatus(plantId: Int)
 }
 
 class PlantsRepositoryImpl(
-    private val localDataSource: PlantsDataSource
-): PlantsRepository {
+    private val localDataSource: PlantsDataSource,
+    private val dbDataSource: DbDataSource,
+    private val coroutineScope: CoroutineScope
+) : PlantsRepository {
     override fun getPlants(): List<Plant> {
```

Comment:
Let this function be a suspend function to avoid coroutine scope usage

Reply:
Done